streaming/base/dataset.py

```diff
@@ -1148,7 +1148,13 @@ def _ready_thread(self, it: _Iterator) -> None:
 
             # Wait for the shard for this sample to be downloaded and decompressed, if not already.
             shard_id, _ = self.spanner[sample_id]
-            self.download_shard(shard_id, True)
+            # During cold shard eviction, shard state might go in a reverse direction. If a shard
```

Comment:
the reverse

Reply:
https://github.com/mosaicml/streaming/pull/279/commits/7ec42c2973e0da7b2f3062a673fcdc78fadaac90